### DISCUSSION TITLE
Names for LeafDB.app.ConceptSqlSet entries

### DISCUSSION THREAD

**Discussion by artgoldberg** [uwrit/leaf#506] [Ideas]
Hello All

I think it would be handy to have text names for `LeafDB.app.ConceptSqlSet` entries. They can be optional.

My motivating example accesses the [`cdm.observation`](https://ohdsi.github.io/CommonDataModel/cdm53.html#OBSERVATION) table. Its description says:

> The OBSERVATION table captures clinical facts about a Person obtained in the context of examination, questioning or a procedure. Any data that cannot be represented by any other domains, such as social and lifestyle facts, medical history, family history, etc. are recorded here.

Thus, while `cdm.observation` contains a date and a `visit_occurrence_id` field, the observation may be a time-invariant property of the person it describes.

Our deployment now has two `ConceptSqlSet` entries for `cdm.observation`, one that's encounter based and one that isn't. The latter is used to list sets of patients in other clinical repositories:

> INSERT INTO LeafDB.app.ConceptSqlSet (SqlSetFrom, IsEncounterBased, IsEventBased, SqlFieldDate, Created, CreatedBy, Updated, UpdatedBy)
SELECT *
FROM (VALUES ('omop.cdm_deid.person',               @no,  @no, NULL,                               GETDATE(), @user, GETDATE(), @user),
             -- skipping 
             ('omop.cdm_deid.observation',          @yes, @no, '@.observation_datetime',           GETDATE(), @user, GETDATE(), @user),
             -- Create a ConceptSqlSet for Patient cohorts that accesses omop.cdm_deid.observation and is not encounter-based:
             ('omop.cdm_deid.observation',          @no,  @no, NULL,                               GETDATE(), @user, GETDATE(), @user),
             -- skipping 
     ) AS X(col1,col2,col3,col4,col5,col6,col7,col8);

I'd like to give distinct descriptive names to these 2 `ConceptSqlSet`s.

Arthur

**Comment by ndobb** [comment]
Hi @artgoldberg , great idea. We do something similar at UW for several tables/views for the same reason.

Do you have a name you'd suggest for the new field? Also character length? (I'm mostly curious whether this should be a long-hand description-type column, or just a "nickname" for the SQL set, and thus shorter)

Best,
-nic

**Comment by artgoldberg** [comment]
thanks @ndobb . I recommend the field be called `Description` and that it be around 200 characters. That's plenty for my description above. I'm thinking of a "long-hand description-type column" that can be NULL.
Arthur
